omp/matrix/ell_kernels.cpp

```diff
@@ -82,36 +88,38 @@ void spmv(std::shared_ptr<const OmpExecutor> exec,
 
 #pragma omp parallel for
     for (size_type row = 0; row < a->get_size()[0]; row++) {
-        for (size_type j = 0; j < c->get_size()[1]; j++) {
-            c->at(row, j) = zero<OutputValueType>();
-        }
+        std::array<arithmetic_type, num_rhs> partial_sum;
+        partial_sum.fill(zero<arithmetic_type>());
         for (size_type i = 0; i < num_stored_elements_per_row; i++) {
             auto val = a_vals(row + i * stride);
             auto col = a->col_at(row, i);
-            for (size_type j = 0; j < c->get_size()[1]; j++) {
-                c->at(row, j) += val * b_vals(col, j);
+#pragma unroll
+            for (size_type j = 0; j < num_rhs; j++) {
+                partial_sum[j] += val * b_vals(col, j);
             }
         }
+#pragma unroll
+        for (size_type j = 0; j < num_rhs; j++) {
+            [&] { c->at(row, j) = out(row, j, partial_sum[j]); }();
```

Comment:
Could you add a comment somewhere in the files or maybe in the developer guidelines/known issues so that we remember this is due to `icpc` + `openmp`? It looks like something to check periodically whether it's fixed and to remember for other similar codes.

Reply:
I think Intel basically confirmed that they will not be fixing this bug (since they no longer maintain icpc), so we will have to keep this around until we drop support.